realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -672,6 +692,19 @@ public Builder readOnly() {
             return this;
         }
 
+        /**
+         * Sets compactOnLaunch to determine if it should be compacted before returned to the user. It is passed the
+         * total file size (data + free space) and the total bytes used by data in the file.
+         *
+         * @param compactOnLaunch a callback called when opening a Realm for the first time during the life of a process
+         *                        to determine if it should be compacted before being returned to the user. It is passed
+         *                        the total file size (data + free space) and the total bytes used by data in the file.
+         */
+        public Builder compactOnLaunch(CompactOnLaunchCallback compactOnLaunch) {
+            this.compactOnLaunch = compactOnLaunch;
```

Comment:
I think we should have a check if `compactOnLaunch` is not null.
If users pass `null` as callback, current implementation does not compact file at all.